src/preferenceCenter/preferenceCenter.commands.js

```diff
@@ -0,0 +1,183 @@
+   * human readable texts/slots/component.
+   * {dynamiccontent} => Dynamic Content
+   */
+  convertPreferenceCenterTokensToSlots() {
+    const components = this.pcService.getDcComponents();
+
+    components.forEach((comp) => {
+      this.pcService.transformDcTokenToSlot(comp);
+    });
+  }
+
+  /**
+   * Convert dynamic content slots to tokens
+   * Dynamic Content => {preferenceCenter}
+   *
+   * @param editor
+   * @returns integer nr of dynamic content slots found
+   */
+  // eslint-disable-next-line class-methods-use-this
+  convertPreferenceCenterSlotsToTokens(editor) {
+    // get all dynamic content elements loaded in the editor
+    const preferenceCenter = editor.DomComponents.getWrapper().find('[data-slot="preferenceCenter"]');
+    preferenceCenter.forEach((preferenceCenter) => {
+      const attributes = preferenceCenter.getAttributes();
+      const decId = attributes['data-param-dec-id'];
+      // If it's not a token -> convert to token
+      if (decId >= 0) {
+        const dynConId = PreferenceCenterCommands.getPcStoreId(attributes['data-param-dec-id']);
+
+        const dynConTarget = mQuery(dynConId);
+        const dynConName = dynConTarget.find(`${dynConId}_tokenName`).val();
+        const dynConToken = `{dynamiccontent="${dynConName}"}`;
+
+        // Clear id because it's reloaded by Mautic and this prevent slot to be destroyed by GrapesJs destroy event on close.
+        // preferenceCenter.addAttributes({ 'data-param-dec-id': '' });
+        preferenceCenter.set('content', dynConToken);
+      }
+    });
+
+    return preferenceCenter.length;
+  }
+
+  /**
+   * Build and display the Dynamic Content editor popup/modal window.
+   * @param {Model} component The current grapesjs component
+   */
+  // eslint-disable-next-line class-methods-use-this
+  showPreferenceCenterEditor(editor, options) {
+    this.dcPopup = PreferenceCenterCommands.buildPreferenceCenterPopup();
+
+    this.addPreferenceCenterEditor(editor, options);
+
+    const title = Mautic.translate('grapesjsbuilder.preferenceCenterBlockLabel');
+    const modal = editor.Modal;
+    modal.setTitle(title);
+
+    editor.Modal.setContent(this.dcPopup);
+    modal.open();
+    modal.onceClose(() => editor.stopCommand('preset-mautic:dynamic-content-open'));
+  }
+
+  /**
+   * Build the basic popup/modal frame to hold the dynamic content editor
+   * @returns HTMLDivElement
+   */
+  static buildPreferenceCenterPopup() {
+    const content = document.createElement('div');
+    content.setAttribute('id', 'dynamic-content-popup');
+
+    const codePopup = document.createElement('div');
+    codePopup.appendChild(content);
+
+    return codePopup;
+  }
+
+  /**
+   * Load Dynamic Content editor and append to the codePopup Modal
+   */
+  addPreferenceCenterEditor(editor, options) {
+    const { target } = options;
+    const dcComponent = target || editor.getSelected();
+    if (!dcComponent) {
+      throw new Error('No dc components found');
```

Comment:
should be changed to "No preference center components found"

Reply:
this file is removed as there is no use as of now

Reply:
please remove the file then